streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1741,6 +1761,21 @@ class ZStream[-R, +E, +A] private[stream] (private[stream] val structure: ZStrea
     }
   }
 
+  /**
+   * Interrupts the evaluation of this stream when the provided IO completes. The given
+   * IO will be forked as part of this stream, and its success will be discarded. This
+   * combinator will also interrupt any in-progress element being pulled from upstream.
+   *
+   * If the IO completes with a failure, the stream will emit that failure.
+   */
+  final def interruptWhen[E1 >: E](io: IO[E1, Unit]): ZStream[R, E1, A] =
```

Comment:
This can be `[R1 <: R, E1 >: E](zio: ZIO[R1, E1, Any])`.

It is actually idiomatic in ZIO to use `Any` when the result is not required - that saves the callers an extra call to `map`, resulting in (slightly) more efficient and concise code :-)

Reply:
Understood – done!